File: combat/src/classicthunder/cards/DefendCard.java
package classicthunder.cards;

import classicthunder.combat.character.CharacterStats;
import classicthunder.combat.character.CharacterType;
import com.badlogic.gdx.graphics.Texture;

public class DefendCard extends Card {

    public DefendCard(Texture texture) {
        super(texture, 1);
    }

    @Override
    public int GetEnergyCost() {
        return 1;
    }

    @Override
    public boolean CanApplyEffects(CharacterStats stats) {
        return (stats.getCharacterType() == CharacterType.PLAYER);
    }

    @Override
    public void ApplyEffects(CharacterStats stats) {
        stats.AdjustBlock(5);
    }
}

File: deck/src/classicthunder/card/DefendCard.kt
package classicthunder.card

import classicthunder.character.CharacterStats
import classicthunder.character.CharacterType
import com.badlogic.gdx.graphics.Texture

class DefendCard(texture: Texture) : Card(texture)
{
    override fun getEnergyCost(): Int {
        return 1
    }

    override fun canApplyEffects(stats: CharacterStats): Boolean {
        return stats.characterType == CharacterType.PLAYER
    }

    override fun applyEffects(stats: CharacterStats) {
        stats.AdjustBlock(5)
    }
}